fields/evaluate.go

```diff
@@ -21,113 +21,190 @@ import (
 		return []byte("count"), nil
-	case AggregateMin:
+	case TimeAggregationMin:
 		return []byte("min"), nil
-	case AggregateMax:
+	case TimeAggregationMax:
 		return []byte("max"), nil
-	case AggregateSum:
+	case TimeAggregationSum:
 		return []byte("sum"), nil
-	case AggregateAvg:
+	case TimeAggregationAvg:
 		return []byte("avg"), nil
-	case AggregateStateHistSeconds:
+	case TimeAggregationSeconds:
 		return []byte("state-histogram-seconds"), nil
-	case AggregateStateHistPercent:
+	case TimeAggregationPercent:
 		return []byte("state-histogram-percent"), nil
-	case AggregateStateHistRate:
+	case TimeAggregationRate:
 		return []byte("state-histogram-rate"), nil
 	}
-	return nil, fmt.Errorf("unknown aggregation method")
+	return nil, fmt.Errorf("bad aggregation method")
 }
 
-func (m *AggregateMethod) UnmarshalText(data []byte) error {
+func (m *TimeAggregationMethod) UnmarshalText(data []byte) error {
 	switch string(data) {
 	case "":
-		*m = AggregateDefault
+		*m = TimeAggregationDefault
 	case "count":
-		*m = AggregateCount
+		*m = TimeAggregationCount
 	case "min":
-		*m = AggregateMin
+		*m = TimeAggregationMin
 	case "max":
-		*m = AggregateMax
+		*m = TimeAggregationMax
 	case "sum":
-		*m = AggregateSum
+		*m = TimeAggregationSum
 	case "avg":
-		*m = AggregateAvg
+		*m = TimeAggregationAvg
 	case "state-histogram-seconds":
-		*m = AggregateStateHistSeconds
+		*m = TimeAggregationSeconds
 	case "state-histogram-percent":
-		*m = AggregateStateHistPercent
+		*m = TimeAggregationPercent
 	case "state-histogram-rate":
-		*m = AggregateStateHistRate
+		*m = TimeAggregationRate
 	default:
 		return fmt.Errorf("bad aggregation method")
 	}
 	return nil
 }
 
-type ItemAggregation struct {
-	Alias       string          `json:"alias,omitempty"`
-	ID          string          `json:"id,omitempty"`
-	Aggregation AggregateMethod `json:"aggregation,omitempty"`
-	State       int             `json:"state"`
-	Lead        int             `json:"lead,omitempty"`
-	Lag         int             `json:"lag,omitempty"`
+func (m GroupAggregation) MarshalText() ([]byte, error) {
+	switch m {
+	case GroupAggregationDefault:
+		return nil, nil
+	case GroupAggregationCount:
+		return []byte("count"), nil
+	case GroupAggregationMin:
+		return []byte("min"), nil
+	case GroupAggregationMax:
+		return []byte("max"), nil
+	case GroupAggregationSum:
+		return []byte("sum"), nil
+	case GroupAggregationAvg:
+		return []byte("avg"), nil
+	}
+	return nil, fmt.Errorf("bad aggregation method")
 }
 
-var _ json.Marshaler = ItemAggregation{}
+func (m *GroupAggregation) UnmarshalText(data []byte) error {
+	switch string(data) {
+	case "":
+		*m = GroupAggregationDefault
+	case "count":
+		*m = GroupAggregationCount
+	case "min":
+		*m = GroupAggregationMin
+	case "max":
+		*m = GroupAggregationMax
+	case "sum":
+		*m = GroupAggregationSum
+	case "avg":
+		*m = GroupAggregationAvg
+	default:
+		return fmt.Errorf("bad aggregation method")
+	}
+	return nil
+}
+
+type Item struct {
```

Comment:
We can not name this filed simply "Item", as it can be confused to mean a Clarify Item (resource). Instead, this is referring specifically to the _items_ field on the evaluate method.

We need to call it something else. `EvaluateItem` could work better than `ItemAggregation`.

```suggestion
type EvaluateItem struct {
```